src/java/org/apache/cassandra/io/sstable/Descriptor.java

```diff
@@ -260,14 +265,14 @@ public static Descriptor fromFilename(File file)
         if (!Version.validate(versionString))
             throw invalidSSTable(name, "invalid version %s", versionString);
 
-        int generation;
+        SSTableId generation;
         try
         {
-            generation = Integer.parseInt(tokens.get(1));
+            generation = SSTableIdFactory.instance.fromString(tokens.get(1));
         }
-        catch (NumberFormatException e)
+        catch (RuntimeException e)
         {
-            throw invalidSSTable(name, "the 'generation' part of the name doesn't parse as a number");
+            throw invalidSSTable(name, "the 'generation' part of the name doesn't parse as a valid unique identifier");
```

Comment:
I think it would be:
```suggestion
            throw invalidSSTable(name, "the 'id' part of the name doesn't parse as a valid unique identifier");
```